### ISSUE TITLE
command peacock.* not found

### ISSUE THREAD

**Issue by dance2die** [johnpapa/vscode-peacock#88]
# command peacock.* not found

Running Peacock via `Show all commands: ctrl+shift+p` fails to run
and shows `command peacock.{all peacock commands} not found` alert message.

![demo](https://user-images.githubusercontent.com/8465237/54164742-2b7b4d80-4434-11e9-9aa9-2c56572c3cb2.gif)

## Version Information

> Windows 10
![winver](https://user-images.githubusercontent.com/8465237/54164387-c4a96480-4432-11e9-841d-12f9df4e8d5a.png)

> VS Code version and whether it is stable or insiders?
![vscode version](https://user-images.githubusercontent.com/8465237/54164413-e276c980-4432-11e9-98ec-b7da46b2d876.png)

> Version of Peacock
1.2.0

## Repro steps
1. Install Peacock
1. Restart VS Code
1. Run any Peacock commands using "ctrl+shift+p"

![all commands ](https://user-images.githubusercontent.com/8465237/54164480-2964bf00-4433-11e9-814f-3be7c289919c.png)

## The log given by the failure

I get following error popups for every command
![error popups](https://user-images.githubusercontent.com/8465237/54164515-4b5e4180-4433-11e9-98b2-19bf38bc1655.png)

## Mention any other details that might be useful

I've also tried this on Ubuntu but the error occurs the same way.

![Linux demo](https://user-images.githubusercontent.com/8465237/54164812-71d0ac80-4434-11e9-99f4-378abe8bc2b2.gif)

![vscode](https://user-images.githubusercontent.com/8465237/54164618-a7c16100-4433-11e9-83bb-c4416640536f.png)

**Comment by cervantn** [comment]
This issue also occurs for me on macOS. Details:

* macOS Sierra 10.12.6
* Code 1.32.1
* Peacock 1.2.0

**Comment by twhite96** [comment]
Same here. macOS 10.14.3. Also, Peacock does not affect the status or activity bars.

* Code 1.32.1

**Comment by johnpapa** [comment]
I’ll takr a look tonight. For now revert to 1.1.0 if you like.

**Comment by johnpapa** [comment]
Hmm ... all tests pass. The CI build and tests pass. I can run it in debug mode and it works.


But if I package it, it fails to run. I need to look deeper

**Comment by dance2die** [comment]
In case it helps, VS Code debug console shows following errors

![vscode devtool console error messages](https://user-images.githubusercontent.com/8465237/54167022-4783ec80-443e-11e9-9502-59816e4f4aa4.png)

and clicking on `notificationsAlerts.ts:38` opens following file.

![notificationsAlerts.ts:38](https://user-images.githubusercontent.com/8465237/54167076-85811080-443e-11e9-8f9d-824a61f6d1a6.png)

**Comment by johnpapa** [comment]
Thanks. 

It isn't finding any of the commands in the CommandRegistry. But again, only when packaged. If I run it in debug mode, it's fine.

Very odd.

**Comment by johnpapa** [comment]
Y'all can find 1.1.0 here, until I figure out the fix


https://marketplace.visualstudio.com/_apis/public/gallery/publishers/johnpapa/vsextensions/vscode-peacock/1.1.0/vspackage

**Comment by dance2die** [comment]
1.1.0 works well~

**Comment by johnpapa** [comment]
I'll track some notes here. It's like it can't figure out where the handlers are for the commands. 

```
  commands.registerCommand(Commands.changeColorToVueGreen, () => {
    console.log('hello');
    //changeColorToVueGreenHandler();
    console.log('bye');
  });
```

If I run the code above, it console logs them out, no problems. No errors.

If I uncomment that line of code, then we get the error. I'm checking into npm package changes, typescript compilation issues, and any other changes that may have caused this. It's very confusing

**Comment by johnpapa** [comment]
I found one bug ... and it makes sense. I added a helper function to a file. And it works everywhere locally when i run and test. Which is because it can get to it. And it works in test and in CI, because it can get to it/

When I package it, the test files are not included (by design). And since I mistakenly put this helper function in a file in the test folder, it only worked locally but not when I packaged it. 

So the solution to this bug is to move the function to a production file. Sounds obvious now - but wow.
